rp2040-hal/src/lib.rs

```diff
@@ -37,6 +37,7 @@
 
 #![warn(missing_docs)]
 #![no_std]
+#![feature(naked_functions)]
```

Comment:
Is my understanding correct that this is not stable yet ?

Reply:
Yeah, it's only on nightly. Are we going to use it on stable too?

Reply:
This library targets stable rust, yes. So nightly features can't be used.

Reply:
It seems that even stabilizing naked is not enough. Using compiler_builtins requires nightly.
But we can change it to default weak in rust. I think the key here is still to stabilize the naked first.

Reply:
Could you please help me verify that the current modification works?

Reply:
Sure, what can I do?
Currently, it doesn't build. Obviously not with stable, but I also get errors with nightly:
```
$ cargo +nightly build --release
[...]
   Compiling rp2040-hal v0.9.1 (/home/jan/rp2040/rp-rs/rp-hal/rp2040-hal)
error: invalid instruction
   --> rp2040-hal/src/sio.rs:299:22
    |
299 |                     "hwdivider_head",
    |                      ^
    |
note: instantiated into assembly here
   --> <inline asm>:1:2
    |
1   |     hwdivider_head
    |     ^
[...]
```

Reply:
I'm not sure about this error yet. But I was able to compile it successfully at https://github.com/rp-rs/rp2040-project-template. I also need add `compiler_builtins = { version = "0.1.104", features = ["weak-intrinsics"] }` to the rp2040-project-template.